### PR TITLE
Varnish 5

### PR DESCRIPTION
**Pull request by cleberjsantos**
Tests Refactor with Varnish 4, Varnish 4 old version and Varnish 5
Varnish 5 implementation (Sync and refactor pull request #38)

### CODE REVIEW

**Comment by cleberjsantos** [comment]
Yeeess..., Travis agrees!

**Comment by mauritsvanrees** [comment]
I am confused. Why do you want to merge this into the varnish-5 branch? Shouldn't this be a PR for the master branch?
In case it is not clear: the varnish-5 branch is not meant to be a long-living branch, but is intended to be merged into master once it is good.
I suppose it is best to create a new PR for merging your branch to master.

**Comment by cleberjsantos** [comment]
Oops, yes, you're right, I'm sending you to the master branch. Tks.

File: README.rst [hunk 1]
```diff
@@ -169,11 +169,15 @@ Two parameters are different/ extra:
...
 ``varnish_version``
-    Varnish target version. Default is ``4.1``.  Options are:
+    Varnish target version. Default is ``5.2``.  Options are:
 
     - 4.0: uses 4.0.5, will stick to 4.0.x
     - 4.1: uses 4.1.9, will stick to 4.1.x
     - 4: uses 4.1.9, will stick to 4.x
+    - 5.0: uses 5.0.0, will stick to 5.0.x
+    - 5.1: uses 5.1.3, will stick to 5.1.x
+    - 5.2: uses 5.1.3, will stick to 5.2.x
```
Comment (mauritsvanrees) [inline README.rst]: Should be: '5.2: uses 5.2.1'.

File: README.rst [hunk 2]
```diff
@@ -273,6 +277,10 @@ These options are available for the recipe part plone.recipe.varnish:configurati
     clients. Useful if there is another port mapping in front of varnish, such
     as haproxy. Defaults to 443.
 
+``vcl-version``
+    Varnish VCL format version.
+    If not given it defaults to ``4.0``.
```
Comment (mauritsvanrees) [inline README.rst]: So we have varnish 5.2 as default version, but 4.0 as default vcl-version. Is that correct? Seems strange at first sight. If it is correct, you may want to explain a bit, maybe say which varnish versions are compatible with which vcl-versions.

Comment (cleberjsantos) [reply README.rst]: Yes :) i'm look here and fixing

About VCL version, although the Varnish version's 4 or 5, we are talking about the VCL syntax version (https://book.varnish-software.com/4.0/chapters/VCL_Basics.html#vcl-syntax).

I'm adding this parameter (vcl-version) foreseeing a near future, so, when the VCL syntax version has a new release, we can say which one we want to use.

**Comment by cleberjsantos** [comment]
And... Finally, green again :smile:

**Comment by jensens** [comment]
Finally, a `CHANGES.rst` entry is missing and then, IMO, this one can be merged immediately.

**Comment by cleberjsantos** [comment]
Tks @mauritsvanrees , @jensens ... Done, waiting travis :)

**Comment by mauritsvanrees** [comment]
Merged. Thanks a lot!

**Comment by mauritsvanrees** [comment]
I have released 2.2.0 with this.

**Comment by jensens** [comment]
Thanks a lot for this change!

**Pull request merged**